### PR TITLE
dom/met signature: `make_base_laplace`

### PR DESCRIPTION
**Pull request by Shoeboxam**
Adds `input_domain` and `input_metric` as arguments to `make_base_laplace`, and removes the `D` type argument from python APIs. 

This adds support for partial construction:
`prev_trans >> part_base_laplace(scale)`

The `D` type argument is filled in from `prev_trans`.

### CODE REVIEW

**Comment by Shoeboxam** [comment]
Current dependencies on/for this PR:
* main
  * **PR #696** <a href="https://app.graphite.dev/github/pr/opendp/opendp/696" target="_blank">Graphite</a> 
    * **PR #645** <a href="https://app.graphite.dev/github/pr/opendp/opendp/645" target="_blank">Graphite</a> 
      * **PR #604** <a href="https://app.graphite.dev/github/pr/opendp/opendp/604" target="_blank">Graphite</a> 
        * **PR #699** <a href="https://app.graphite.dev/github/pr/opendp/opendp/699" target="_blank">Graphite</a> 
        * **PR #706** <a href="https://app.graphite.dev/github/pr/opendp/opendp/706" target="_blank">Graphite</a> 
        * **PR #687** <a href="https://app.graphite.dev/github/pr/opendp/opendp/687" target="_blank">Graphite</a> 
          * **PR #688** <a href="https://app.graphite.dev/github/pr/opendp/opendp/688" target="_blank">Graphite</a> 
            * **PR #689** <a href="https://app.graphite.dev/github/pr/opendp/opendp/689" target="_blank">Graphite</a> 
              * **PR #712** <a href="https://app.graphite.dev/github/pr/opendp/opendp/712" target="_blank">Graphite</a>  👈
              * **PR #702** <a href="https://app.graphite.dev/github/pr/opendp/opendp/702" target="_blank">Graphite</a> 
                * **PR #703** <a href="https://app.graphite.dev/github/pr/opendp/opendp/703" target="_blank">Graphite</a> 
                  * **PR #704** <a href="https://app.graphite.dev/github/pr/opendp/opendp/704" target="_blank">Graphite</a> 
                    * **PR #705** <a href="https://app.graphite.dev/github/pr/opendp/opendp/705" target="_blank">Graphite</a> 
              * **PR #690** <a href="https://app.graphite.dev/github/pr/opendp/opendp/690" target="_blank">Graphite</a> 
                * **PR #512** <a href="https://app.graphite.dev/github/pr/opendp/opendp/512" target="_blank">Graphite</a> 
                * **PR #692** <a href="https://app.graphite.dev/github/pr/opendp/opendp/692" target="_blank">Graphite</a> 
                  * **PR #697** <a href="https://app.graphite.dev/github/pr/opendp/opendp/697" target="_blank">Graphite</a> 
                  * **PR #514** <a href="https://app.graphite.dev/github/pr/opendp/opendp/514" target="_blank">Graphite</a> 
      * **PR #691** <a href="https://app.graphite.dev/github/pr/opendp/opendp/691" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/opendp/opendp/712?utm_source=stack-comment).

File: rust/src/measurements/laplace/ffi.rs [hunk 1]
```diff
@@ -1,36 +1,44 @@
...
+        let input_metric = try_!(input_metric.downcast_ref::<D::InputMetric>()).clone();
         let scale = *try_as_ref!(scale as *const D::Atom);
-        make_base_laplace::<D>(scale, Some(k)).into_any()
+        make_base_laplace::<D>(input_domain, input_metric, scale, Some(k)).into_any()
     }
+    let input_domain = try_as_ref!(input_domain);
+    let input_metric = try_as_ref!(input_metric);
     let k = k as i32;
-    let D = try_!(Type::try_from(D));
+    let D = input_domain.type_.clone();
```
Comment (andrewvyrros) [inline rust/src/measurements/laplace/ffi.rs:38]: It's cool that `suppress` lets you get rid of the `D` arg and do this.

**Review by andrewvyrros — APPROVED** [review]
Looks great!

I expect there will be a little churn in this one from changes in #689, will revisit then.

File: docs/source/examples/unknown-dataset-size.ipynb [hunk 2]
```diff
@@ -95,8 +95,9 @@
    ],
    "source": [
     "from opendp.transformations import *\n",
-    "from opendp.measurements import make_base_laplace\n",
-    "from opendp.mod import enable_features\n",
+    "from opendp.measurements import part_base_laplace, make_base_discrete_laplace\n",
+    "from opendp.mod import enable_features, binary_search_chain\n",
```
Comment (andrewvyrros) [inline docs/source/examples/unknown-dataset-size.ipynb:99]: `binary_search_chain` is imported below in the cell where you use it.

File: docs/source/examples/unknown-dataset-size.ipynb [hunk 3]
```diff
@@ -95,8 +95,9 @@
...
+    "from opendp.domains import atom_domain\n",
```
Comment (andrewvyrros) [inline docs/source/examples/unknown-dataset-size.ipynb:100]: `atom_domain` is imported below with `vector_domain`.

File: rust/src/measurements/laplace/mod.rs [hunk 4]
```diff
@@ -119,29 +119,39 @@ where
     Ok((k, relaxation))
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "derive"))]
```
Comment (andrewvyrros) [inline rust/src/measurements/laplace/mod.rs:122]: Should this be `feature = "partials"`?

**Review by andrewvyrros — APPROVED** [review]

**Review by andrewvyrros — APPROVED** [review]

**Pull request merged**